test/e2e/scrapeconfig_test.go

```diff
@@ -1065,3 +1068,78 @@ var EC2SDTestCases = []scrapeCRDTestCase{
 		expectedError: true,
 	},
 }
+
+var staticConfigTestCases = []scrapeCRDTestCase{
+	{
+		name: "Valid targets",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+
+			StaticConfigs: []monitoringv1alpha1.StaticConfig{
+				{
+					Targets: []monitoringv1alpha1.Target{"1.1.1.1:9090", "0.0.0.0:9090"},
+				},
+			},
+		},
+		expectedError: false,
+	},
+	{
+		name: "Invalid absent targets",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+
+			StaticConfigs: []monitoringv1alpha1.StaticConfig{
+				{},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid duplicate targets",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+
+			StaticConfigs: []monitoringv1alpha1.StaticConfig{
+				{
+					Targets: []monitoringv1alpha1.Target{"1.1.1.1:9090", "1.1.1.1:9090"},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Invalid empty targets",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+
+			StaticConfigs: []monitoringv1alpha1.StaticConfig{
+				{
+					Targets: []monitoringv1alpha1.Target{},
+				},
+			},
+		},
+		expectedError: true,
+	},
+	{
+		name: "Valid labels",
+		scrapeConfigSpec: monitoringv1alpha1.ScrapeConfigSpec{
+
+			StaticConfigs: []monitoringv1alpha1.StaticConfig{
+				{
+					Targets: []monitoringv1alpha1.Target{"1.1.1.1:9090", "0.0.0.0:9090"},
+					Labels:  map[monitoringv1.LabelName]string{"owned-by": "prometheus"},
```

Comment:
"owned-by" isn't valid.

Reply:
I see what you mean. However, the test does not fail nor does the cluster fail to accept this object when deployed locally. What should we do?
One thing I can think of is try making the Labels a struct with a `listType=Map` and `listKey=LabelName` and see if that works.

Reply:
But then, how would the JSON look like. I don't quite understand this.

Reply:
> I see what you mean. However, the test does not fail nor does the cluster fail to accept this object when deployed locally. What should we do?

https://groups.google.com/g/kubebuilder/c/CMvdWa4et_k?pli=1

Indeed the only option *would* be to use `+listType=map` but it would be against the idea that ScrapeConfig is a 1:1 mapping of the Prometheus config format...

As such we'd need a new PR to change the `map[v1.LabelName]string` type to `map[string]string` and validate the keys in the resource selector.